### ISSUE TITLE
Bug: Create tags error

### ISSUE THREAD

**Issue by jerviscui** [platers/obsidian-linter#472]
## Describe the Bug

When there is a tag in the file and the front also has a tag, duplicate tags are generated and there are problems with the YAML format.

## How to Reproduce

1. My Linter settings:
![image](https://user-images.githubusercontent.com/6396748/200128431-e8db6562-d33f-4a34-9d75-ce18b69a77b9.png)
![image](https://user-images.githubusercontent.com/6396748/200128438-dc0f7838-c96c-490d-90bc-1a5e220de313.png)

2. The file's tags like this,
![image](https://user-images.githubusercontent.com/6396748/200128363-ebad7a44-032d-4d47-b52f-f667df4ee9bc.png)

3. Then use Linter to format the file,
![image](https://user-images.githubusercontent.com/6396748/200128389-d0b03261-74ef-4929-84ef-c2d2eb9683fe.png)

4. Question 2: If tags end with a space, like 

	```
	tags: [moc, roadmap, test1, test2, test3 todo doing, test3 ]
	```

5. The latest tag will add again.
![image](https://user-images.githubusercontent.com/6396748/200128895-ea8ed054-ab80-4a18-9b84-eca6ca116591.png)

## Expected Behavior

Should probably be added without duplicating the tag and have the correct formatting.

```markdown
tags: [moc, roadmap, test1, test2, test3, todo, doing]
```

## Device

- [x] Desktop
- [ ] Mobile

**Comment by pjkaufman** [comment]
Thanks for reporting this. Could you provide a reproducible example? I don't see where to do and doing are as tags.

**Comment by pjkaufman** [comment]
Also, if the image in 2) is your starting format, then the results of the lint look correct to me. It is not valid in YAML to use both commas and spaces as separators. So even though it may seem link they would be separated, in YAML they don't count as new entries.

**Comment by jerviscui** [comment]
> Thanks for reporting this. Could you provide a reproducible example? I don't see where to do and doing are as tags.

@pjkaufman Sorry. I've updated the issue, not sure if this clarifies the issue. My English may not be very good.

**Comment by jerviscui** [comment]
There may be a future based on this issue. 
Can the corresponding tag in YAML be automatically deleted when the tag is deleted from the article content?

Maybe it could be implemented like this `linter-yaml-title-alias` function.

**Comment by pjkaufman** [comment]
> There may be a future based on this issue. Can the corresponding tag in YAML be automatically deleted when the tag is deleted from the article content?
> 
> Maybe it could be implemented like this `linter-yaml-title-alias` function.

This would have to be another FR since it is a bit different. The only problem is this gets really tricky if you have tags that are not in the YAML front matter that you want to keep.

**Comment by pjkaufman** [comment]
As for the originalissue, only the second scenario is a bug. The first one is not going to work because it is an invalid use of YAML.

**Comment by jerviscui** [comment]
> As for the original issue, only the second scenario is a bug. The first one is not going to work because it is an invalid use of YAML.

Yes, it just happened when I was in the process of editing the file.

**Comment by jerviscui** [comment]
I modified the description of question 2 again, and indeed spaces before and after can cause problems.

**Comment by pjkaufman** [comment]
So the following is what I plan to do for this bug:
- trim the start and end of values in the tags array as they are parsed to make sure duplicates are not added

This fix should be relatively simple. Once it is merged, I will see about going ahead and creating a bug fix release.

The issue with trying to use multiple formats described in question 1 is not really something that can be fixed in a realistic manner as it would likely make things very error prone and it goes against how YAML works.

As for the request to add a feature to keep the tags in the document and the tags in the frontamtter in sync, that will have to be added as a separate issue if this is desired.

**Comment by pjkaufman** [comment]
This issue should be resolved on master and the next release. Please let us know if this is not resolved on either master or the next release (I plan to release the bug fixes today or tomorrow). Feel free to put in the feature request for the keeping of tags in sync with the body content.
